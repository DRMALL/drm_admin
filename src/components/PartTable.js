

import React from 'react'
import Table from 'antd/lib/table'
import { browserHistory } from 'react-router'
import TableTextHidden from './TableTextHidden'

const PartTable = ({ table }) => {
  const columns =[{
    title: '物料代码',
    dataIndex: 'code',
  },{
    title: '物料名称',
    dataIndex: 'name',
  },{
    title: '规格型号',
    dataIndex: 'model',
  },{
    title: '件／台／套',
    dataIndex: 'reserve',
  },{
    title: '类型',
    dataIndex: 'types',
  },{
    title: '备注',
    dataIndex: 'remark',
    render: text => TableTextHidden(text, 30)
  },{
    title: '添加时间',
    dataIndex: 'time',
  },{
    title: '操作',
    key: 'action',
    render: (text, record) =>
    <div data-part-id={ record._id } onClick={ getPartId } >
      <i className="iconfont icon-compile trouble-table-icon"  style={{ color: '#0068d2' }} ></i>
    </div>
  }]
  return (
    <Table dataSource={ table } columns={ columns } rowKey='_id' />
    )
}

function getPartId(e){
  localStorage.setItem('partId', e.currentTarget.dataset.partId )
  browserHistory.push('/parts/remark')
}

export default PartTable
