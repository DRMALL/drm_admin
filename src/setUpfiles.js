
//mock localStorage和react 的，只需要在文件中引入就行
import React from 'react';
if (typeof window !== 'undefined') {
    const matchMediaPolyfill = function matchMediaPolyfill() {
    return {
      matches: false,
      addListener() {
      },
      removeListener() {
      },
    };
  };
  window.matchMedia = window.matchMedia || matchMediaPolyfill;

  window.React = React;
  window.localStorage = ( function storageMock() {
    var storage = {};
    return {
      setItem: function(key, value) {
        storage[key] = value || '';
      },
      getItem: function(key) {
        return key in storage ? storage[key] : null;
      },
      removeItem: function(key) {
        delete storage[key];
      },
      get length() {
        return Object.keys(storage).length;
      },
      key: function(i) {
        var keys = Object.keys(storage);
        return keys[i] || null;
      }
    }
  })()
}
