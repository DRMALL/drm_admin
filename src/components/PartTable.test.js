

import React from 'react'
import PartTable from './PartTable'
import setUpfiles from '../setUpfiles'
import { render } from 'enzyme'
import toJson from 'enzyme-to-json'

it('PartTable test', () => {
  const table = [{_id: 'sdd344', code:'d3560', name:'物料名称', model: '5dd43', },
                 {_id: 'sdd34d5', code:'d3546', name:'物料名称', model: '5dds5', }]
  const wrapper = render(<PartTable table={ table } /> )
  expect(toJson(wrapper)).toMatchSnapshot()
})
