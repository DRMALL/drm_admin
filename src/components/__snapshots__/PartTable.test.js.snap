// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PartTable test 1`] = `
<div
  class="ant-table-wrapper"
>
  <div
    class="ant-spin-nested-loading"
  >
    <div
      class="ant-spin-container"
    >
      <div
        class="ant-table ant-table-large ant-table-scroll-position-left"
      >
        <div
          class="ant-table-content"
        >
          <div
            class="ant-table-body"
          >
            <table
              class=""
            >
              <colgroup>
                <col />
                <col />
                <col />
                <col />
                <col />
                <col />
                <col />
                <col />
              </colgroup>
              <thead
                class="ant-table-thead"
              >
                <tr>
                  <th
                    class=""
                  >
                    <span>
                      物料代码
                    </span>
                  </th>
                  <th
                    class=""
                  >
                    <span>
                      物料名称
                    </span>
                  </th>
                  <th
                    class=""
                  >
                    <span>
                      规格型号
                    </span>
                  </th>
                  <th
                    class=""
                  >
                    <span>
                      件／台／套
                    </span>
                  </th>
                  <th
                    class=""
                  >
                    <span>
                      类型
                    </span>
                  </th>
                  <th
                    class=""
                  >
                    <span>
                      备注
                    </span>
                  </th>
                  <th
                    class=""
                  >
                    <span>
                      添加时间
                    </span>
                  </th>
                  <th
                    class=""
                  >
                    <span>
                      操作
                    </span>
                  </th>
                </tr>
              </thead>
              <tbody
                class="ant-table-tbody"
              >
                <tr
                  class="ant-table-row  ant-table-row-level-0"
                >
                  <td
                    class=""
                  >
                    <span
                      class="ant-table-row-indent indent-level-0"
                      style="padding-left:0px;"
                    />
                    d3560
                  </td>
                  <td
                    class=""
                  >
                    物料名称
                  </td>
                  <td
                    class=""
                  >
                    5dd43
                  </td>
                  <td
                    class=""
                  />
                  <td
                    class=""
                  />
                  <td
                    class=""
                  />
                  <td
                    class=""
                  />
                  <td
                    class=""
                  >
                    <div
                      data-part-id="sdd344"
                    >
                      <i
                        class="iconfont icon-compile trouble-table-icon"
                        style="color:#0068d2;"
                      />
                    </div>
                  </td>
                </tr>
                <tr
                  class="ant-table-row  ant-table-row-level-0"
                >
                  <td
                    class=""
                  >
                    <span
                      class="ant-table-row-indent indent-level-0"
                      style="padding-left:0px;"
                    />
                    d3546
                  </td>
                  <td
                    class=""
                  >
                    物料名称
                  </td>
                  <td
                    class=""
                  >
                    5dds5
                  </td>
                  <td
                    class=""
                  />
                  <td
                    class=""
                  />
                  <td
                    class=""
                  />
                  <td
                    class=""
                  />
                  <td
                    class=""
                  >
                    <div
                      data-part-id="sdd34d5"
                    >
                      <i
                        class="iconfont icon-compile trouble-table-icon"
                        style="color:#0068d2;"
                      />
                    </div>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
      <ul
        class="ant-pagination ant-table-pagination"
        unselectable="unselectable"
      >
        <li
          aria-disabled="true"
          class="ant-pagination-disabled ant-pagination-prev"
          tabindex="0"
          title="上一页"
        >
          <a />
        </li>
        <li
          class="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
          tabindex="0"
          title="1"
        >
          <a>
            1
          </a>
        </li>
        <li
          aria-disabled="true"
          class="ant-pagination-disabled ant-pagination-next"
          tabindex="0"
          title="下一页"
        >
          <a />
        </li>
      </ul>
    </div>
  </div>
</div>
`;
